docs/Avoiding-Disruptions.md

```diff
@@ -0,0 +1,36 @@
+# Avoiding Disruptions
+
+Under certain conditions, MachineConfig updates will cause the Machine Config
+Daemon to drain and/or reboot nodes affected by the change. This doc aims to
+identify what changes are disruptive as well as the ways in which they are
+disruptive. 
+
+This is particularly important for non-high-availability clusters, where a
+drain and reboot can be more disruptive.
+
+## Non-Disruptive Updates
+
+Updating the following files will occur without a drain and reboot:
+- `/etc/kubernetes/kubelet-ca.crt`
+- `/var/lib/kubelet/config.json`
+- [SSH key updates](https://github.com/openshift/machine-config-operator/blob/ca8fcb887f153604556f2706cee8a33165879797/docs/Update-SSHKeys.md)
+
+## Drains
+
+Updates to the following files will cause node(s) to be drained:
+- `/etc/containers/registries.conf`
+
+While the node will not reboot, it will drain to allow
+[CRI-O](https://cri-o.io) to reload. 
+
+## Reboots
+
+The following changes will cause node(s) to drain and reboot:
+
+- [OS Updates](https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigDaemon.md#os-updates)
+- [KernelArguments](https://github.com/openshift/machine-config-operator/blob/ca8fcb887f153604556f2706cee8a33165879797/docs/MachineConfiguration.md#kernelarguments)
+- [systemd units](https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigDaemon.md#systemd-unit-updates)
+- [KernelType](https://github.com/openshift/machine-config-operator/blob/ca8fcb887f153604556f2706cee8a33165879797/docs/MachineConfiguration.md#kerneltype)
+- [RHCOS Extensions](https://github.com/openshift/machine-config-operator/blob/ca8fcb887f153604556f2706cee8a33165879797/docs/MachineConfiguration.md#rhcos-extensions)
+- [FIPS](https://github.com/openshift/machine-config-operator/blob/ca8fcb887f153604556f2706cee8a33165879797/docs/MachineConfiguration.md#fips). It should be noted that unlike other items on this list, FIPS is not a toggleable option.
```

Comment:
As @darkmuggle pointed out earlier, if FIPS is not a day 2 operation does it really cause a disruption?

Reply:
It's not, I'll drop this line entirely.